### PR TITLE
Shravan/cus 1216 add requestlog tables to explorer

### PR DESCRIPTION
**Pull request by ShravanCool**

### CODE REVIEW

**Review by shailykhamesra — APPROVED** [review]

File: explorer/utils.py [hunk 1]
```diff
@@ -49,6 +49,11 @@ def get_connection_pii():
     return connections[app_settings.EXPLORER_CONNECTION_PII_NAME] if app_settings.EXPLORER_CONNECTION_PII_NAME else connection
 
 
+def get_connection_asyncapi_db():
+    logger.info("Connecting with async-api DB")
+    return connections[app_settings.EXPLORER_CONNECTION_ASYNC_API_DB_NAME] if app_settings.EXPLORER_CONNECTION_ASYNC_API_DB_NAME else connection
```
Comment (Ujjwal-Squadstack) [inline explorer/utils.py:55]: Why are we returning the connection in else part? 🤔

File: explorer/utils.py [hunk 2]
```diff
@@ -284,4 +289,12 @@ def check_replication_lag():
 
     return True, human(replication_lag, 4)
 
-    
+
+def should_route_to_asyncapi_db(sql):
+    request_log_tables = ["request_log_requestlog", "request_log_requestlogdata"]
+    pattern = r"\b(?:%s)\b" % "|".join(map(re.escape, request_log_tables))
```
Comment (Ujjwal-Squadstack) [inline explorer/utils.py:296]: What in case request_log_requestlog2 table is passed within sql?

Comment (ShravanCool) [reply explorer/utils.py:296]: We don't want anyone to access `request_log_requestlog2` table. If they are querying for `requestlog` or `requestlogdata` table, the function returns `True` and we run the query in the async-api db else it returns `False`. The `requestlog2` table queries will be run using the main db connection, which will display the error message it is currently displaying if someone tries to run query with `requestlog2` in it.

File: explorer/utils.py (hunk 1 above)
Comment (ShravanCool) [reply explorer/utils.py:55]: The checks are like this-
1. Check if `is connection_pii` and if yes run query with the PII connection DB
2. Else, check if the query is related to `requestlog` or `requestlogdata` and if yes run query with the asyncapi db connection.
3. Else, the query is related to other tables and thus run with the main explorer connection.

**Review by Ujjwal-Squadstack — APPROVED** [review]

File: explorer/utils.py [hunk 3]
```diff
@@ -284,4 +290,58 @@ def check_replication_lag():
...
+    match = re.search(pattern, sql)
+    if match:
+        return True
+
+    return False
+
+
+def add_cutoff_date_to_requestlog_queries(sql):
```
Comment (Ujjwal-Squadstack) [inline explorer/utils.py:304]: @ShravanCool Can you add a unit case for this?

Comment (ShravanCool) [reply explorer/utils.py:304]: Added test-cases for this method.

**Review by shailykhamesra — APPROVED** [review]

**Review by Ujjwal-Squadstack — APPROVED** [review]

**Pull request merged**